<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>OpenDynamicCreator</class>
 <widget class="QMainWindow" name="OpenDynamicCreator">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1051</width>
    <height>595</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OpenDynamic Creator</string>
  </property>
  <property name="locale">
   <locale language="English" country="UnitedKingdom"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QVBoxLayout" name="layoutCentralWidget">
      <property name="spacing">
       <number>0</number>
      </property>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1051</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionLoad"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuTracks">
    <property name="title">
     <string>Tracks</string>
    </property>
    <addaction name="actionAddTrack"/>
    <addaction name="actionAddClip"/>
    <addaction name="actionAddGroup"/>
    <addaction name="separator"/>
    <addaction name="actionDeleteMusicItem"/>
   </widget>
   <widget class="QMenu" name="menuWindow">
    <property name="title">
     <string>Window</string>
    </property>
    <addaction name="actionMusicWindow"/>
    <addaction name="actionEventsWindow"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
    <addaction name="separator"/>
    <addaction name="actionPreferences"/>
   </widget>
   <widget class="QMenu" name="menuEvents">
    <property name="title">
     <string>Events</string>
    </property>
    <addaction name="actionAddEvent"/>
    <addaction name="actionAddEventCommand"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuTracks"/>
   <addaction name="menuEvents"/>
   <addaction name="menuWindow"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusMain"/>
  <widget class="QDockWidget" name="dockMusic">
   <property name="allowedAreas">
    <set>Qt::LeftDockWidgetArea|Qt::RightDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>Music</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockMusicContents">
    <layout class="QVBoxLayout" name="verticalLayout_5">
     <item>
      <widget class="MusicTreeView" name="treeMusic">
       <property name="contextMenuPolicy">
        <enum>Qt::CustomContextMenu</enum>
       </property>
       <property name="acceptDrops">
        <bool>true</bool>
       </property>
       <property name="dragEnabled">
        <bool>true</bool>
       </property>
       <property name="dragDropMode">
        <enum>QAbstractItemView::DragDrop</enum>
       </property>
       <property name="defaultDropAction">
        <enum>Qt::MoveAction</enum>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::ExtendedSelection</enum>
       </property>
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectRows</enum>
       </property>
       <property name="headerHidden">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="toolbarPlayback">
   <property name="windowTitle">
    <string>Toolbar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionPlayPause"/>
   <addaction name="actionStop"/>
  </widget>
  <widget class="QDockWidget" name="dockEvents">
   <property name="windowTitle">
    <string>Events</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockEventsContents">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="EventTreeView" name="treeEvents">
       <property name="headerHidden">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionPlayPause">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/play</normaloff>
     <normalon>:/icons/pause</normalon>:/icons/play</iconset>
   </property>
   <property name="text">
    <string>Play/Pause</string>
   </property>
   <property name="toolTip">
    <string>Play/Pause</string>
   </property>
   <property name="shortcut">
    <string>Space</string>
   </property>
  </action>
  <action name="actionStop">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/stop</normaloff>:/icons/stop</iconset>
   </property>
   <property name="text">
    <string>Stop</string>
   </property>
   <property name="toolTip">
    <string>Pause the currently playing song.</string>
   </property>
   <property name="shortcut">
    <string>Esc</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="text">
    <string>New</string>
   </property>
   <property name="toolTip">
    <string>Create a new project and close the current project.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionLoad">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="toolTip">
    <string>Open a saved project.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="toolTip">
    <string>Save the current project.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="text">
    <string>Save As...</string>
   </property>
   <property name="toolTip">
    <string>Save the current project as a new file.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="toolTip">
    <string>Exit the application.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+?</string>
   </property>
  </action>
  <action name="actionAddTrack">
   <property name="text">
    <string>Add Music</string>
   </property>
   <property name="toolTip">
    <string>Add Music</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+N</string>
   </property>
  </action>
  <action name="actionMusicWindow">
   <property name="text">
    <string>Music</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F1</string>
   </property>
  </action>
  <action name="actionEventsWindow">
   <property name="text">
    <string>Events</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F2</string>
   </property>
  </action>
  <action name="actionAddClip">
   <property name="text">
    <string>Add Clip</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+C</string>
   </property>
  </action>
  <action name="actionAddGroup">
   <property name="text">
    <string>Add Mixer</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+M</string>
   </property>
  </action>
  <action name="actionDeleteMusicItem">
   <property name="text">
    <string>Delete</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="text">
    <string>Undo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="text">
    <string>Redo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Z</string>
   </property>
  </action>
  <action name="actionAddEvent">
   <property name="text">
    <string>Add Event</string>
   </property>
   <property name="toolTip">
    <string>Add Event</string>
   </property>
  </action>
  <action name="actionAddEventCommand">
   <property name="text">
    <string>Add Event Command</string>
   </property>
   <property name="toolTip">
    <string>Add Event Command</string>
   </property>
  </action>
  <action name="actionPreferences">
   <property name="text">
    <string>Preferences</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MusicTreeView</class>
   <extends>QTreeView</extends>
   <header>Widgets/MusicTreeView.h</header>
  </customwidget>
  <customwidget>
   <class>EventTreeView</class>
   <extends>QTreeView</extends>
   <header>Widgets/EventTreeView.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionAbout</sender>
   <signal>triggered()</signal>
   <receiver>OpenDynamicCreator</receiver>
   <slot>showAboutDialogue()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>585</x>
     <y>326</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAddTrack</sender>
   <signal>triggered()</signal>
   <receiver>OpenDynamicCreator</receiver>
   <slot>addTrack()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>594</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAddClip</sender>
   <signal>triggered()</signal>
   <receiver>OpenDynamicCreator</receiver>
   <slot>addClipFileDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>595</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAddGroup</sender>
   <signal>triggered()</signal>
   <receiver>OpenDynamicCreator</receiver>
   <slot>addClipGroup()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>595</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionMusicWindow</sender>
   <signal>triggered()</signal>
   <receiver>dockMusic</receiver>
   <slot>showNormal()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>136</x>
     <y>268</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>treeMusic</sender>
   <signal>customContextMenuRequested(QPoint)</signal>
   <receiver>OpenDynamicCreator</receiver>
   <slot>customContextMenuMusic(QPoint)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>136</x>
     <y>222</y>
    </hint>
    <hint type="destinationlabel">
     <x>595</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDeleteMusicItem</sender>
   <signal>triggered()</signal>
   <receiver>OpenDynamicCreator</receiver>
   <slot>deleteMusicItem()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>595</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAddEvent</sender>
   <signal>triggered()</signal>
   <receiver>OpenDynamicCreator</receiver>
   <slot>addEventCommand()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>602</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAddEventCommand</sender>
   <signal>triggered()</signal>
   <receiver>OpenDynamicCreator</receiver>
   <slot>addEvent()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>602</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>OpenDynamicCreator</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>525</x>
     <y>297</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPlayPause</sender>
   <signal>toggled(bool)</signal>
   <receiver>OpenDynamicCreator</receiver>
   <slot>playSong(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>525</x>
     <y>297</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStop</sender>
   <signal>triggered(bool)</signal>
   <receiver>OpenDynamicCreator</receiver>
   <slot>stopSong()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>525</x>
     <y>297</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionEventsWindow</sender>
   <signal>triggered()</signal>
   <receiver>dockEvents</receiver>
   <slot>showNormal()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>913</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setExtraConditionValueVisible(int)</slot>
  <slot>setConditionsEnabled(int)</slot>
  <slot>showTracksWindow()</slot>
  <slot>showLoopSettingsWindow()</slot>
  <slot>showStatesWindow()</slot>
  <slot>showAboutDialogue()</slot>
  <slot>addTrack()</slot>
  <slot>addClip()</slot>
  <slot>addClipGroup()</slot>
  <slot>customContextMenuMusic(QPoint)</slot>
  <slot>deleteMusicItem()</slot>
  <slot>undo()</slot>
  <slot>redo()</slot>
  <slot>addEvent()</slot>
  <slot>addEventCommand()</slot>
  <slot>playSong(bool)</slot>
  <slot>stopSong()</slot>
  <slot>addClipFileDialog()</slot>
 </slots>
</ui>
